torch/_meta_registrations.py

```diff
@@ -7011,6 +7011,54 @@ def meta_local_scalar_dense(self: Tensor):
     raise RuntimeError("Tensor.item() cannot be called on meta tensors")
 
 
+@register_meta(aten.silu)
+def silu(self: Tensor) -> Tensor:
+    return torch.empty_like(self)
+
+
+@register_meta(aten.sigmoid)
+def sigmoid(self: Tensor) -> Tensor:
+    return torch.empty_like(self)
+
+
+@register_meta(aten._softmax)
+def softmax(x: Tensor, dim: int, half_to_float: bool) -> Tensor:
+    if half_to_float:
+        assert x.dtype == torch.half
+    computation_dtype, result_dtype = utils.elementwise_dtypes(
+        x, type_promotion_kind=utils.ELEMENTWISE_TYPE_PROMOTION_KIND.DEFAULT
+    )
+
+    res = torch.empty_like(x)
+    res = res.contiguous()
+    result_dtype = result_dtype if not half_to_float else computation_dtype
+    res = res.to(result_dtype)
```

Comment:
```suggestion
    result_dtype = result_dtype if not half_to_float else computation_dtype
    res = torch.empty_like(x, dtype=result_dtype, memory_format=torch.contiguous_format)
```
No need for extra copies here: just create the empty tensor right the first time.